### PR TITLE
Refactor all the logging to use pf::log instead of Log::Log4perl directly

### LINKED ISSUE #754: Move to pf::log
Remove all direct references to Log::Log4perl move to pf::log

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Refactor all the logging to use pf::log instead of Log::Log4perl directly
## Impacts

Should be none
## NEWS file entries

New Features
++++++++++++

Enhancements
++++++++++++
- Refactoring and code cleanup

Bug Fixes
+++++++++
## Fixes

Fixes #754
## Delete branch after merge

NO

### CODE REVIEW

**Comment by jrouzierinverse** [comment]
Review me

File: html/captive-portal/lib/captiveportal/PacketFence/Controller/Activate/Email.pm [hunk 1]
```diff
@@ -4,7 +4,7 @@ use namespace::autoclean;
 
 BEGIN { extends 'captiveportal::Base::Controller'; }
 
-use Log::Log4perl;
+use pf::log;
```
Comment (whitx) [inline html/captive-portal/lib/captiveportal/PacketFence/Controller/Activate/Email.pm]: Double call of pf::log

File: lib/pf/Authentication/Condition.pm [hunk 2]
```diff
@@ -97,7 +98,7 @@ sub matches {
             }
         }
         else {
-            my $logger = Log::Log4perl->get_logger( __PACKAGE__ );
+            my $logger = get_logger();
```
Comment (whitx) [inline lib/pf/Authentication/Condition.pm:101]: unnecessary my $logger has been defined earlier

File: lib/pf/ConfigStore.pm [hunk 3]
```diff
@@ -17,7 +17,7 @@ Is the Base class for accessing pf::config::cached
 use Moo;
 use namespace::autoclean;
 use pf::config::cached;
-use Log::Log4perl qw(get_logger);
+use pf::log;
```
Comment (whitx) [inline lib/pf/ConfigStore.pm:20]: This file calls 'get_logger' directly w/o a 'my $logger' definition, need to add it

File: lib/pf/Portal/Session.pm [hunk 4]
```diff
@@ -24,7 +24,7 @@ use CGI::Session::Driver::chi;
 use pf::CHI;
 use HTML::Entities;
 use Locale::gettext qw(bindtextdomain textdomain bind_textdomain_codeset);
-use Log::Log4perl;
+use pf::log;
```
Comment (whitx) [inline lib/pf/Portal/Session.pm:27]: Double call of pf::log

File: lib/pf/Switch/Cisco/SG300.pm [hunk 5]
```diff
@@ -23,20 +23,22 @@ Developed and tested on SG300 running 1.1.2.0
...
+=back
+
 =cut
 
 use strict;
 use warnings;
 
 use base ('pf::Switch::Cisco::Catalyst_2960');
-use Log::Log4perl;
+use pf::log;
```
Comment (whitx) [inline lib/pf/Switch/Cisco/SG300.pm:36]: unecessary call of pf::log

**Comment by rouzier** [comment]
Please re-review

File: lib/pf/WebAPI.pm [hunk 6]
```diff
@@ -11,7 +11,7 @@ use warnings;
 
 use Apache2::MPM ();
 use Apache2::RequestRec;
-use Log::Log4perl;
+use pf::log;
```
Comment (whitx) [inline lib/pf/WebAPI.pm:14]: This file still have multiple use of Log::Log4perl

File: lib/pf/action.pm [hunk 7]
```diff
@@ -117,7 +117,7 @@ sub action_exist {
 
 sub action_add {
     my ($vid, $action) = @_;
-    my $logger = Log::Log4perl::get_logger('pf::action');
+    my $logger = get_logger;
     if ( action_exist( $vid, $action ) ) {
         $logger->warn("attempt to add existing action $action to class $vid");
```
Comment (whitx) [inline lib/pf/action.pm:123]: missing brackets

File: lib/pf/action.pm [hunk 8]
```diff
@@ -148,7 +148,7 @@ sub action_view_all {
 
 sub action_delete {
     my ($vid, $action) = @_;
-    my $logger = Log::Log4perl::get_logger('pf::action');
+    my $logger = get_logger;
     db_query_execute(ACTION, $action_statements, 'action_delete_sql', $vid, $action) || return (0);
     $logger->debug("action $action deleted from class $vid");
```
Comment (whitx) [inline lib/pf/action.pm:154]: missing brackets

File: lib/pf/action.pm [hunk 9]
```diff
@@ -157,7 +157,7 @@ sub action_delete {
 
 sub action_delete_all {
     my ($vid) = @_;
-    my $logger = Log::Log4perl::get_logger('pf::action');
+    my $logger = get_logger;
     db_query_execute(ACTION, $action_statements, 'action_delete_all_sql', $vid) || return (0);
     $logger->debug("all actions for class $vid deleted");
```
Comment (whitx) [inline lib/pf/action.pm:163]: missing brackets

File: lib/pf/action.pm [hunk 10]
```diff
@@ -167,7 +167,7 @@ sub action_delete_all {
 # TODO what is that? Isn't it dangerous?
 sub action_api {
     my ($mac, $vid) = @_;
-    my $logger = Log::Log4perl::get_logger('pf::action');
+    my $logger = get_logger;
     my $class_info = class_view($vid);
     my @params = split(' ', $class_info->{'external_command'});
```
Comment (whitx) [inline lib/pf/action.pm:173]: missing brackets

File: lib/pf/action.pm [hunk 11]
```diff
@@ -189,7 +189,7 @@ sub action_api {
 
 sub action_execute {
     my ($mac, $vid, $notes) = @_;
-    my $logger = Log::Log4perl::get_logger('pf::action');
+    my $logger = get_logger;
     my $leave_open = 0;
     my @actions = class_view_actions($vid);
```
Comment (whitx) [inline lib/pf/action.pm:195]: missing brackets

File: lib/pf/action.pm [hunk 12]
```diff
@@ -283,7 +283,7 @@ sub action_email {
 
 sub action_log {
     my ($mac, $vid) = @_;
-    my $logger = Log::Log4perl::get_logger('pf::action');
+    my $logger = get_logger;
     require pf::iplog;
     my $ip = pf::iplog::mac2ip($mac) || 0;
```
Comment (whitx) [inline lib/pf/action.pm:321]: missing brackets

File: lib/pf/action.pm [hunk 13]
```diff
@@ -374,7 +374,7 @@ sub action_autoregister {
 
 sub action_close {
```
Comment (whitx) [inline lib/pf/action.pm]: missing brackets

File: lib/pf/action.pm [hunk 14]
```diff
@@ -383,7 +383,7 @@ sub action_close {
 
    if (defined($class->{'vclose'})) {
```
Comment (whitx) [inline lib/pf/action.pm]: missing brackets

File: lib/pf/floatingdevice.pm [hunk 15]
```diff
@@ -82,24 +82,24 @@ use pf::util;
...
+
+=item new
 
 Get an instance of the pf::floatingdevice object
 
 =cut
 
 sub new {
-    my $logger = Log::Log4perl::get_logger("pf::floatingdevice");
+    my $logger = get_logger;
```
Comment (whitx) [inline lib/pf/floatingdevice.pm]: Missing parathenses

File: lib/pf/floatingdevice.pm [hunk 16]
```diff
@@ -109,14 +109,14 @@ sub enablePortConfig {
 
 # FIXME
 # we have to change the error handling in case we leave the function before we enable Linkup/down traps
-# cause in this case there would be any traps enabled anymore... 
+# cause in this case there would be any traps enabled anymore...
 
 
     my ($this, $mac, $switch, $switch_port, $switch_locker_ref, $radius_triggered) = @_;
-    my $logger = Log::Log4perl::get_logger('pf::floatingdevice');
+    my $logger = get_logger;
```
Comment (whitx) [inline lib/pf/floatingdevice.pm]: Missing parathenses

File: lib/pf/floatingdevice.pm [hunk 17]
```diff
@@ -169,10 +169,10 @@ sub disablePortConfig {
 
 # FIXME
 # we have to change the error handling in case we leave the function before we enable port-security traps
-# cause in this case there would be any traps enabled anymore... 
+# cause in this case there would be any traps enabled anymore...
 
     my ($this, $mac, $switch, $switch_port, $switch_locker_ref) = @_;
-    my $logger = Log::Log4perl::get_logger('pf::floatingdevice');
+    my $logger = get_logger;
```
Comment (whitx) [inline lib/pf/floatingdevice.pm]: Missing parathenses

File: lib/pf/floatingdevice.pm [hunk 18]
```diff
@@ -245,9 +245,9 @@ Puts the switchport in MAB floating device mode
 
 sub enableMABFloating{
     my ( $this, $mac, $switch, $ifIndex ) = @_;
-    my $logger = Log::Log4perl::get_logger('pf::floatingdevice');
-     
-    my $result;         
+    my $logger = get_logger;
```
Comment (whitx) [inline lib/pf/floatingdevice.pm]: Missing parathenses

File: lib/pf/floatingdevice.pm [hunk 19]
```diff
@@ -268,12 +268,12 @@ Verifies if there is a floating device plugged into the switchport in the locati
 
 sub portHasFloatingDevice {
     my ($this, $switch, $switch_port) = @_;
-    my $logger = Log::Log4perl::get_logger('pf::floatingdevice');
+    my $logger = get_logger;
```
Comment (whitx) [inline lib/pf/floatingdevice.pm]: Missing parathenses

File: lib/pf/floatingdevice.pm [hunk 20]
```diff
@@ -291,7 +291,7 @@ Disconnects the active locationlog macs on the port so they reauthenticate to be
 
 sub _disconnectCurrentDevices{
     my ( $this, $switch, $switch_port ) = @_;
-    my $logger = Log::Log4perl::get_logger('pf::floatingdevice');
+    my $logger = get_logger;
```
Comment (whitx) [inline lib/pf/floatingdevice.pm]: Missing parathenses

File: addons/database-cleaner.pl [hunk 21]
```diff
@@ -32,16 +32,8 @@ =head1 DESCRIPTION
...
-use DBI;
-use Data::Dumper;
-use Getopt::Long;
-use pf::log;
-use Pod::Usage;
-
 BEGIN {
-  use Log::Log4perl;
-  use pf::log();
+  use Log::Log4perl qw(get_logger);
```
Comment (whitx) [inline addons/database-cleaner.pl:36]: I thought that was Log4perl remove process?

File: bin/cluster/sync [hunk 22]
```diff
@@ -33,20 +33,8 @@ use constant INSTALL_DIR => '/usr/local/pf';
...
-use pf::file_paths;
-use pf::cluster;
-use fingerbank::FilePath;
-use pf::log;
-use Pod::Usage;
-
 BEGIN {
-  use Log::Log4perl;
-  use pf::log();
+  use Log::Log4perl qw(get_logger);
```
Comment (whitx) [inline bin/cluster/sync:37]: still has Log4perl

**Comment by whitx** [comment]
Rebase please

File: addons/database-cleaner.pl (hunk 21 above)
Comment (rouzier) [inline addons/database-cleaner.pl:36]: The original author wants to bypass the standard pf logging so in this case I am making it explicit that is what is happening.

**Pull request merged**